Only link libxml2 statically in purely static build

We now link libxml2 dynamically if both shared and static builds were
requested. This avoids overlinking and should fix #36.

https://gitlab.gnome.org/GNOME/libxslt/-/issues/36
https://gitlab.gnome.org/GNOME/libxslt/-/commit/616b2418f7aaf453dac4bd5de5bc59a7ac8584e2
--- configure.orig	2019-10-30 15:02:02.000000000 -0500
+++ configure	2022-09-15 02:14:18.000000000 -0500
@@ -14707,7 +14707,7 @@
 
 
 if test "x$LIBXML_CONFIG_PREFIX" = "x" -a "x$LIBXML_LIBS" = "x"; then
-    if test "$build_static_libs" = "no"; then
+    if test "$build_static_libs" = "yes"; then
 
 pkg_failed=no
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for LIBXML" >&5
